contracts/feeders/RebasedFeederPool.sol

```diff
@@ -0,0 +1,928 @@
+
+        Asset memory input_;
+        for (uint256 i = 0; i < len; i++) {
+            input_ = _getAsset(_assets[i]);
+            indexes[i] = input_.idx;
+            require(input_.exists, "Invalid asset");
+
+            for (uint256 j = i + 1; j < len; j++) {
+                require(_assets[i] != _assets[j], "Duplicate asset");
+            }
+        }
+    }
+
+    /**
+     * @dev Gets all config needed for general InvariantValidator calls
+     */
+    function _getConfig() internal view returns (FeederConfig memory) {
+        return FeederConfig(totalSupply() + data.pendingFees, _getA(), data.weightLimits);
+    }
+
+    /**
+     * @dev Gets current amplification var A
+     */
+    function _getA() internal view returns (uint256) {
+        AmpData memory ampData_ = data.ampData;
+
+        uint64 endA = ampData_.targetA;
+        uint64 endTime = ampData_.rampEndTime;
+
+        // If still changing, work out based on current timestamp
+        if (block.timestamp < endTime) {
+            uint64 startA = ampData_.initialA;
+            uint64 startTime = ampData_.rampStartTime;
+
+            (uint256 elapsed, uint256 total) = (block.timestamp - startTime, endTime - startTime);
+
+            if (endA > startA) {
+                return startA + (((endA - startA) * elapsed) / total);
+            } else {
+                return startA - (((startA - endA) * elapsed) / total);
+            }
+        }
+        // Else return final value
+        else {
+            return endA;
+        }
+    }
+
+    /**
+     * @dev Get actual basset data with updated vault balance of fAsset if liquidity index changed.
+     */
+    function _getMemBassetData() internal view returns (BassetData[] memory bAssetData) {
+        bAssetData = new BassetData[](NUM_ASSETS);
+        bAssetData[M_INDEX] = data.bAssetData[M_INDEX];
+        bAssetData[F_INDEX].vaultBalance = uint128(IERC20(data.bAssetPersonal[F_INDEX].addr).balanceOf(address(this)));
+        bAssetData[F_INDEX].ratio = data.bAssetData[F_INDEX].ratio;
+    }
+
+    /***************************************
+                    YIELD
+    ****************************************/
+
+    /**
+     * @dev Collects the interest generated from the lending markets, performing a theoretical mint, which
+     * is then validated by the interest validator to protect against accidental hyper inflation.
+     * @return mintAmount   fpToken units generated from interest collected from lending markets
+     * @return newSupply    fpToken total supply after mint
+     */
+    function collectPlatformInterest()
+        external
+        override
+        onlyInterestValidator
+        whenInOperation
+        nonReentrant
+        returns (uint256 mintAmount, uint256 newSupply)
+    {
+        _updateBassetData();
+        (uint8[] memory idxs, uint256[] memory gains) = FeederManager.calculatePlatformInterest(
+            data.bAssetPersonal,
+            data.bAssetData
```

Comment:
- data.bAssetData read was changed for a method where the balance of the vault is updated, except in the method is that correct?